python/sbp/acquisition.py

```diff
@@ -295,9 +333,16 @@ def from_binary(self, d):
     the message.
 
     """
-    p = MsgAcqResultDepB._parser.parse(d)
-    for n in self.__class__.__slots__:
-      setattr(self, n, getattr(p, n))
+    try:
+      self._from_binary(d)
+    except:
+      print traceback.print_exc()
+
+  def __getitem__(self, item):
+    return getattr(self, item)
+
+  def _get_embedded_type(self, t):
+    return globals()[t]
```

Comment:
do we need to use globals() here? I think it would be better to define our type mapping on the message superclass.